src/peripherals.rs

```diff
@@ -44,6 +46,8 @@ pub struct Peripherals {
     pub spi1: spi::SPI1,
     #[cfg(not(feature = "riscv-ulp-hal"))]
     pub spi2: spi::SPI2,
+    pub spi22: spi2::SPI2,
```

Comment:
This is very confusing. What does `spi22` mean? What does `spi2v2` mean?

Also:
- you need `#[cfg(not(feature = "riscv-ulp-hal"))]` or this code will break when compiling for the ULP HAL
- How can I have 3 different SPI2 peripherals? What happens if I use all three simultaneously? As in `spi2` `spi22` and `spi_v2`? Would that even work?

Reply:
yeah sorry to confuse you here. it was just a hack to outsource the current dev of the driver. after finishing the new driver api i will merge it into the spi.rs than this will be all gone. ( and you don't need to review a wall of code :smile: ). When i  created  the draft i started with  a spi_v2.rs. This is now gone i should have removed it :smile_cat:  I split it now into two files because of two different feauteres: spi2.rs witch is mostly an update to the current spi.rs will bring Mutli Cs / Multi Device. spi2_pool is the new thing where you can share rtos device handles with multiple devices to allow more than the limit of 3 or 6 devices, or alternativly just to have an mechanism to seemingly switch already created device_configs on an device( one device needs different bus speeds like an sdcard on creation).